repose-aggregator/src/docs/asciidoc/filters/rate-limiting.adoc

```diff
@@ -1,52 +1,339 @@
+            <limit next-available="2012-06-22T15:38:17.956Z" unit="MINUTE" remaining="10" value="10" verb="GET"/>
+        </rate>
+    </rates>
+</limits>
+----
+
+=== Full Feature Utilization Example
+This configuration provides per-user limits, global limits, and an endpoint to query limit data.
 
 [source,xml]
-.???.cfg.xml
+.rate-limiting.cfg.xml
 ----
-<??? xmlns="http://docs.openrepose.org/repose/???/v1.0">
-    <???> <!--1-->
-    </???>
-</???>
+<rate-limiting xmlns="http://docs.openrepose.org/repose/rate-limiting/v1.0"
+    overLimit-429-responseCode="true" <!--1-->
+    datastore="distributed/hash-ring" <!--2-->
+    datastore-warn-limit="1200" <!--3-->
+    use-capture-groups="true"> <!--4-->
+
+    <request-endpoint <!--5-->
+        include-absolute-limits="true" <!--6-->
+        uri-regex="/limits/?"/> <!--7-->
+
+    <global-limit-group> <!--8-->
+        <limit
+            id="global-resource-per-minute" <!--9-->
+            uri="a global resource" <!--10-->
+            uri-regex="/global/resource/?" <!--11-->
+            http-methods="GET" <!--12-->
+            value="100" <!--13-->
+            unit="SECOND" <!--14-->
+            query-param-names="filter"/> <!--15-->
+    </global-limit-group>
+
+    <limit-group <!--16-->
+        id="customer-limits" <!--17-->
+        groups="group-one group-two" <!--18-->
+        default="true"> <!--19-->
+
+        <limit
+            id="user-resource-one"
+            uri="users resource one"
+            uri-regex="/users/one/([^/]*)/?"
+            http-methods="GET"
+            unit="MINUTE"
+            value="10"/> <!--20-->
+
+        <limit
+            id="user-resource-two"
+            uri="users resource two"
+            uri-regex="/users/two/[^/]*/?"
+            http-methods="POST"
+            unit="DAY"
+            value="2"/> <!--21-->
+    </limit-group>
+</rate-limiting>
 ----
-<1> Specifies ???.
+<1> Specifies whether this filter should return a 413 or 429 response status code when a request is limited.
+<2> Specifies which <<../services/datastores.adoc#, datastore>> should be used.
+<3> Specifies the upper bound for the number of user-specific limits to store in the datastore before a warning is logged.
+<4> Specifies whether or not the content captured by regular expression capture groups should distinguish multiple, separate limits.
+<5> Specifies that the this filter should provide an endpoint where limit data can be queried.
+<6> Specifies that the origin service will supply absolute limits which this filter will use to enrich the response to limit data queries.
+<7> Specifies the regular expression used (i.e., matched against the URI) to determine whether or not a request is a limit data query request.
+<8> Specifies <<Global Limits>>, which are applied irrespective of user.
+<9> Specifies a unique ID which identifies this limit.
+No two limits may have the same ID within a limit group, including the global limit group.
```

Comment:
Pretty sure you're right, but I think we fixed that. The XML schema does not assert that they must be globally unique, and the limit key now uses both.

Ah, here's the commit where we changed it:
https://github.com/rackerlabs/repose/commit/415950eec543e130ad0c499469228235b235601f